firebase-installations/src/main/java/com/google/firebase/installations/remote/FirebaseInstallationServiceClient.java

```diff
@@ -0,0 +1,191 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.installations.remote;
+
+import androidx.annotation.NonNull;
+import java.io.IOException;
+import java.net.URL;
+import java.util.zip.GZIPOutputStream;
+import javax.net.ssl.HttpsURLConnection;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+/** Http client that sends request to Firebase Installations backend API. */
+public class FirebaseInstallationServiceClient {
+  private static final String FIREBASE_INSTALLATIONS_API_DOMAIN =
+      "firebaseinstallations.googleapis.com";
+  private static final String CREATE_REQUEST_RESOURCE_NAME_FORMAT = "projects/%s/installations";
+  private static final String GENERATE_AUTH_TOKEN_REQUEST_RESOURCE_NAME_FORMAT =
+      "projects/%s/installations/%s/auth:generate";
+  private static final String DELETE_REQUEST_RESOURCE_NAME_FORMAT = "projects/%s/installations/%s";
+  private static final String FIREBASE_INSTALLATIONS_API_VERSION = "v1";
+  private static final String FIREBASE_INSTALLATION_AUTH_VERSION = "FIS_V2";
+
+  private static final String CONTENT_TYPE_HEADER_KEY = "Content-Type";
+  private static final String JSON_CONTENT_TYPE = "application/json";
+  private static final String CONTENT_ENCODING_HEADER_KEY = "Content-Encoding";
+  private static final String GZIP_CONTENT_ENCODING = "gzip";
+
+  public enum Code {
+    OK,
+
+    HTTP_CLIENT_ERROR,
+
+    CONFLICT,
+
+    NETWORK_ERROR,
+
+    SERVER_ERROR,
+
+    UNAUTHORIZED,
+  }
+
+  @NonNull
+  public Code createFirebaseInstallation(
```

Comment:
Here we would have to eventually return the response rather than just the Code. This is the initial structure.

Reply:
@diwu-arete  does this way of reading response seem right? cs/"readResponse(HttpsURLConnection conn)"

Another way could be to use OkHttpClient: firebase-functions/src/main/java/com/google/firebase/functions/FirebaseFunctions.java

Reply:
You code actually doesn't read any content from the response, instead, only reads the response code.
So I recommend (Vlad recommended me, hah) to just use the java class HttpsURLConnection

Although it seems that you write more code by using HttpsURLConnection, but for SDK development, a very important goal is to make the SDK smaller in size, I believe okhttpclient(or some other 3P libraries) will make the SDK much bigger, because they have much more functions than our simple use case.

Reply:
PTAL. Let me know if this is the right way. Thanks.